--- libzrtp-master/src/zrtp_iface_scheduler.c.orig	2012-07-07 20:59:23.000000000 +0200
+++ libzrtp-master/src/zrtp_iface_scheduler.c	2025-02-02 14:04:40.533291114 +0100
@@ -7,6 +7,7 @@
  * Viktor Krykun <v.krikun at zfoneproject.com> 
  */
 
+#define _DEFAULT_SOURCE
 #include "zrtp.h"
 
 #if (defined(ZRTP_USE_BUILTIN_SCEHDULER) && (ZRTP_USE_BUILTIN_SCEHDULER ==1))
--- libzrtp-master/src/zrtp_cache_sqlite_backend.c.orig	2012-07-07 20:59:23.000000000 +0200
+++ libzrtp-master/src/zrtp_cache_sqlite_backend.c	2025-02-02 14:06:18.797190264 +0100
@@ -1,6 +1,8 @@
 /*
  */
 
+#define _XOPEN_SOURCE
+#define _XOPEN_SOURCE_EXTENDED
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
--- libzrtp-master/test/cmockery/cmockery.c.orig	2012-07-07 20:59:23.000000000 +0200
+++ libzrtp-master/test/cmockery/cmockery.c	2025-02-02 14:07:31.270940910 +0100
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#define _POSIX_C_SOURCE 200809L
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
